text/0000-project-error-handling.md

```diff
@@ -43,6 +43,13 @@ Here is a tenative starting point, subject to change:
 - What is the consensus on handling `dyn Error`s? Should it be encouraged or discouraged? Should we look into making `Box<dyn Error...>` implement `Error`?
 
 
+### Identify pain points in error handling today
+
+- Backtrace capture is expensive, but without one it can be difficult to pinpoint the origin of errors
+- unwrap on errors without first converting to a reporting type will often discard relevant information
+- errors printing from main have to assume a prefixed `Error: `, sub par control of output format when printing during termination.
+- Error trait only exposes 3 forms of context, can only represent singly linked lists for chains of errors
```

Comment:
```suggestion
- The `Error` trait only exposes three forms of context, can only represent singly-linked lists for chains of errors
```

Reply:
It's unclear what the pain points are here. 

- Is three too many or not enough?
- Why is a singly-linked list painful?

Reply:
3 is not enough, I'd like to be able to return `SpanTraces`, `std::panic::Location`, and other such types via the error trait, this is a reference to the generic member access RFC.

Singly linked lists can be painful for parsers for example, where you encounter multiple errors at once, and they're all the source for a single higher level error, so your chain of errors looks a little more like a tree. This is also addressed by the generic member access proposal where you could access a `dyn Iter<&(dyn Error + 'static)>`.